src/js/angular/clustermanagement/directives/cluster-configuration/multi-region.directive.js

```diff
@@ -15,7 +23,197 @@ function MultiRegion() {
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.deleting'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.enableSecondaryMode = () => {
+                ModalService.openSimpleModal({
+                    title: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.enable_secondary'),
+                    message: $translate.instant('cluster_management.cluster_configuration_multi_region.confirm.enable_secondary_warning'),
+                    warning: true,
+                    backdrop: 'static',
+                    confirmButtonKey: 'common.ok.btn',
+                    stopPropagation: true
+                }).result.then(() => ModalService.openCustomModal({
+                    title: $translate.instant('cluster_management.cluster_configuration_multi_region.secondary_cluster_settings'),
+                    templateUrl: 'js/angular/clustermanagement/templates/modal/secondary-mode-modal.html',
+                    controller: function($scope, $uibModalInstance, config) {
+                        $scope.rpcAddress = '';
+                        $scope.tag = '';
+                        $scope.title = config.title;
+                        $scope.onClick = ($event) => $event.stopPropagation();
+                        $scope.ok = ($event) => {
+                            $uibModalInstance.close({
+                                primaryNode: $scope.rpcAddress,
+                                tag: $scope.tag
+                            });
+                        };
+                        $scope.cancel = function($event) {
+                            $uibModalInstance.dismiss('cancel');
+                        };
+                    },
+                    size: 'lg',
+                    warning: true,
+                    backdrop: 'static'
+                }).result)
+                    .then((payload) => {
+                        $scope.setLoader(true);
+                        return ClusterRestService.enableSecondaryMode(payload);
+                    })
+                    .then(() => toastr.success($translate.instant('cluster_management.cluster_configuration_multi_region.secondary_enabled')))
+                    .catch((response) => {
+                        if (response === 'cancel') {
+                            return;
+                        }
+                        const msg = getError(response);
+                        toastr.error(msg, $translate.instant('cluster_management.cluster_configuration_multi_region.error.secondary'));
+                    })
+                    .finally(() => $scope.setLoader(false));
+            };
+
+            $scope.disableSecondaryMode = () => {
+                $scope.setLoader(true);
```

Comment:
It's too early for this here. Remove it!

Reply:
fixed